### PR TITLE
[LLM] use pytorch linear for large input matrix

### PR DESCRIPTION
**Pull request by yangw1234**
## Description

use pytorch linear for large input matrix

### CODE REVIEW

**Comment by jason-dai** [comment]
Please test it on both client and server machines.

**Comment by yangw1234** [comment]
tested on both server and client

File: python/llm/src/bigdl/llm/transformers/linear_quant.py [hunk 1]
```diff
@@ -82,6 +86,19 @@ def ggml_convert_quant(tensor: torch.Tensor, qtype: int, convert_shape_only=Fals
...
+def ggml_int4_convert_fp32(tensor: torch.Tensor, weight_shape: tuple, k: int):
+    invalidInputError(tensor.dtype == torch.uint8,
+                      "Input tensor must be uint8")
+    src_ptr = ctypes.c_void_p(tensor.data.data_ptr())
+
+    dst_size = k
+    dst_tensor = torch.empty(weight_shape, dtype=torch.float)
+    dst_ptr = ctypes.c_void_p(dst_tensor.data.data_ptr())
+
+    ggml.ggml_dequantize_q4_0(src_ptr, dst_ptr, k)
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/linear_quant.py:98]: add an issue for supporting other low bit types in future

Comment (yangw1234) [reply python/llm/src/bigdl/llm/transformers/linear_quant.py:98]: added

**Comment by jason-dai** [comment]
Is this PR ready?

**Comment by yangw1234** [comment]
> > Is this PR ready?
> 
> Yes, it is ready.

Need one approving review to merge, though.

**Comment by jason-dai** [comment]
> > Is this PR ready?
> 
> Yes, it is ready.

See my comment above wrt `is_server`

File: python/llm/src/bigdl/llm/transformers/linear_quant.py [hunk 2]
```diff
@@ -201,15 +219,19 @@ def forward(self, x: torch.Tensor):
...
         x0 = self.weight.data
 
-        result = ggml_matmul_src1_x_src0_t(x0, x, self.weight_shape, self.qtype)
-        new_shape = x_shape[:-1] + (self.out_len,)
-        result = result.view(new_shape)
-
-        if self.bias is not None:
-            result += self.bias
+        # todo may need to set a different number on different platforms
+        if IS_SERVER and self.qtype == SYM_INT4 and x_2d.shape[0] >= 96:
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/linear_quant.py:227]: use a constant instead of 96

**Comment by jason-dai** [comment]
Otherwise LGTM

**Review by jason-dai — APPROVED** [review]

**Comment by yangw1234** [comment]
> > > Is this PR ready?
> > 
> > 
> > Yes, it is ready.
> 
> See my comment above wrt `is_server`

working on the changes

**Pull request merged**